library/core/resource-loader/src/main/java/org/quiltmc/qsl/resource/loader/api/ResourceLoader.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.resource.loader.api;
+
+import net.minecraft.resource.ResourceType;
+
+import org.quiltmc.qsl.resource.loader.api.reloader.IdentifiableResourceReloader;
+import org.quiltmc.qsl.resource.loader.impl.ResourceLoaderImpl;
+
+/**
+ * Represents the resource loader. Contains different register methods.
+ */
+public interface ResourceLoader {
```

Comment:
We don't have sealed classes till 17-ga, mark this as NonExtendable with JB annotations